clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/ops/DestroyTitusServerGroupAtomicOperation.groovy

```diff
@@ -16,58 +16,54 @@
+import com.netflix.spinnaker.clouddriver.saga.flow.SagaFlow
+import com.netflix.spinnaker.clouddriver.titus.deploy.actions.DestroyTitusJob
+import com.netflix.spinnaker.clouddriver.titus.deploy.actions.ResolveTitusJobId
 import com.netflix.spinnaker.clouddriver.titus.deploy.description.DestroyTitusServerGroupDescription
+import com.netflix.spinnaker.clouddriver.titus.deploy.handlers.DestroyTitusJobCompletionHandler
+import com.netflix.spinnaker.clouddriver.titus.deploy.handlers.TitusExceptionHandler
 import groovy.util.logging.Slf4j
+import org.jetbrains.annotations.NotNull
 
-@Slf4j
-class DestroyTitusServerGroupAtomicOperation implements AtomicOperation<Void> {
+import javax.annotation.Nonnull
 
-  private static final String PHASE = "DESTROY_TITUS_SERVER_GROUP"
-  private final TitusClientProvider titusClientProvider
-  private final DestroyTitusServerGroupDescription description
+@Slf4j
+class DestroyTitusServerGroupAtomicOperation extends AbstractSagaAtomicOperation<DestroyTitusServerGroupDescription, DeleteServerGroupEvent, Void> {
   private final Collection<DeleteServerGroupEvent> events = []
 
-  DestroyTitusServerGroupAtomicOperation(TitusClientProvider titusClientProvider,
-                                         DestroyTitusServerGroupDescription description) {
-    this.titusClientProvider = titusClientProvider
-    this.description = description
+  DestroyTitusServerGroupAtomicOperation(DestroyTitusServerGroupDescription description) {
+    super(description)
+  }
+
+  @Override
+  protected SagaFlow buildSagaFlow(List priorOutputs) {
+    return new SagaFlow()
+      .then(ResolveTitusJobId.class)
+      .then(DestroyTitusJob.class)
+      .exceptionHandler(TitusExceptionHandler.class)
+      .completionHandler(DestroyTitusJobCompletionHandler.class)
   }
 
   @Override
-  Void operate(List priorOutputs) {
-    task.updateStatus PHASE, "Destroying server group: ${description.serverGroupName}..."
-    TitusClient titusClient = titusClientProvider.getTitusClient(description.credentials, description.region)
-    Job job = titusClient.findJobByName(description.serverGroupName)
-    if (job) {
-      titusClient.terminateJob((TerminateJobRequest) new TerminateJobRequest().withJobId(job.id).withUser(description.user))
-      events << new DeleteServerGroupEvent(
-        TitusCloudProvider.ID, description.credentials.name, description.region, description.serverGroupName
-      )
-      task.updateStatus PHASE, "Successfully issued terminate job request to titus for ${job.id} which corresponds to ${description.serverGroupName}"
-    } else {
-      task.updateStatus PHASE, "No titus job found for ${description.serverGroupName}"
-    }
+  protected void configureSagaBridge(@NotNull @Nonnull SagaAtomicOperationBridge.ApplyCommandWrapper.ApplyCommandWrapperBuilder builder) {
+    builder.initialCommand(
+      ResolveTitusJobId.ResolveTitusJobIdCommand.builder().description(description).build()
```

Comment:
I could avoid the `.description(description)` and actually pass in account/region/serverGroupName/user explicitly.